src/cdk/activation.ts

```diff
@@ -22,9 +25,33 @@ export async function activate(activateArguments: { extensionContext: vscode.Ext
         treeDataProvider: explorer,
         showCollapseAll: true
     })
+
+    // Indicates workspace includes a CDK app and user has expanded the Node
+    view.onDidExpandElement(e => {
+        if (e.element.contextValue === 'awsCdkAppNode') {
+            ext.telemetry.record(getTelemetryEvent('appExpanded'))
```

Comment:
Could we record an event per node object's lifetime instead of each time the node is expanded? I'm wondering if this can give us the same amount of engagement info with less messages.

Reply:
Yeah I was also thinking about the noise:signal ration. I'm not sure how we could do that but I think the suggestion makes sense. Any thoughts on what that would look like?

Reply:
The App Nodes could have a simple boolean field to hold whether or not an expand metric has been recorded. The event could check this, and set it when recording an event